src/pkg/egress/syslog/https_batch.go

```diff
@@ -11,7 +11,8 @@ import (
 	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
 )
 
-const BATCHSIZE = 256 * 1024
+var DefaultBatchSize = 256 * 1024
```

Comment:
Could you please make these two as constants?

Reply:
They cannot be constants, as they need to be set in the Tests.
There seems to be no easy way around doing it like this, so can we not keep it this way?

Reply:
For cases like this we're using the so called `Functional options pattern`. An object is created with default values and then options are applied.Sorry for not mentioning it earlier...
Here is an [example definition](https://github.com/cloudfoundry/loggregator-agent-release/blob/main/src/pkg/clientpool/v2/balancer.go#L10-L24) and [usage](https://github.com/cloudfoundry/loggregator-agent-release/blob/main/src/cmd/loggregator-agent/app/app_v2.go#L173] from this project.

You can find more info about this pattern in the following blog posts:
https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c
https://golang.cafe/blog/golang-functional-options-pattern.html

This way, you will be able to have the default values configured and can be overwritten if needed.